process/block/shardblock.go

```diff
@@ -1107,15 +1113,25 @@ func (sp *shardProcessor) requestMissingFinalityAttestingHeaders() uint32 {
 			sp.dataPool.HeadersNonces())
 
 		if err != nil {
-			requestedBlockHeaders++
-			go sp.onRequestHeaderHandlerByNonce(sharding.MetachainShardId, i)
+			missingFinalityAttestingHeaders++
```

Comment:
you could create a function at baseProcess for lines 1110-1121 and call from here and metaprocessor as well.

Reply:
Extracted/refactored the whole method in baseProcessor